tests/chainerx_tests/unit_tests/routines_tests/test_linalg.py

```diff
@@ -123,9 +123,11 @@ def setup(self):
 class TestSolve(NumpyLinalgOpTest):
 
     def generate_inputs(self):
+        # To control the scale of the output x, we sample (a, x) and compute
+        # b from them instead of sampling (a, b) directly.
         a = numpy.random.random(self.shape).astype(self.dtypes[0])
```

Comment:
This test seems to fail if `a` is singular. (`numpy.linalg.solve` raises a `LinAlgError` in such case.)

Reply:
Ah, you are right, `a` could be singular. OK, I will consider such a case.

Reply:
I realized that there are some matrices often cause numerical errors other than singular matrices (like as Hilbert matrix). It seems difficult to avoid such inputs, and also difficult to allow this numerical error by relaxing the tolerances. I think it's OK to allow stochastic failures by using `condition.repeat_with_success_at_least`. How do you think?